### ISSUE TITLE
[Feature Request] draft request

### ISSUE THREAD

**Issue by TinsFox** [usebruno/bruno#2919]
### I have checked the following:

- [X] I've searched existing issues and found nothing related to my issue.

### Describe the feature you want to add

Please consider adding a draft request. Many times, I may just want to test a request and discard it after the test is completed, so I do not need to save it to the collection.

### Mockups or Images of the feature

<img width="551" alt="image" src="https://github.com/user-attachments/assets/93bf5aa0-ebfe-4147-897f-508f27623553">

**Comment by rajib-raiyat** [comment]
I was looking for something exactly like this. We can implement draft requests by using temporary file storage for the `.bru` file format. [Bruno's technology stack](https://github.com/usebruno/bruno/blob/main/contributing.md#technology-stack) supports this well, and we can use a package like [tmp (Node.js)](https://www.npmjs.com/package/tmp) to handle temporary files seamlessly across Linux, macOS, and Windows.


Here’s a basic implementation:

```js
const tmp = require('tmp');
const fs = require('fs');

const tempFile = tmp.fileSync({ postfix: '.bru' });
fs.writeFileSync(tempFile.name, 'Draft request data');

// Cleanup the temp file when the user closes the tab
tempFile.removeCallback();
```

This approach allows us to store draft requests temporarily, giving users the flexibility to test requests without saving them permanently. The temporary files will be automatically cleaned up when no longer needed.

**Comment by zishraq** [comment]
@TinsFox I think the approach can be simplified by directly opening a `Draft Request` upon clicking `New Request`. This approach (similar to Postman) would allow me to start building the request without immediately needing to provide all the necessary details. Once the request is ready to be saved, Bruno could then prompt me for the required information.

**Comment by Nisgrak** [comment]
@zishraq But the Draft and allow to save without assign to a Collection is a wonderful feature in HTTPie, most of the time I don't want my test to be included in the .json

**Comment by NavSwisscom** [comment]
Hi Guys,
It'll be very helpful to have this feature. Hope it'll be available soon.

**Comment by GautMorel** [comment]
Hello Guys,
Do you have any new for this topic ?
Thank you in advance

**Comment by anusree-bruno** [comment]
Hi everyone,
We really appreciate all the interest and thoughtful suggestions around draft requests — it's clear this is a highly requested feature.

At the moment, we're focused on bringing a few major features to Bruno, but this is definitely on our radar. Once those are completed, we’d love to revisit this and explore how best to implement it. Thank you all for your patience and continued support!

**Comment by ryanthon** [comment]
Is there any news on plans for when this functionality may be added? This is the one thing my team is waiting for to switch to Bruno 😊

**Comment by SamRay1024** [comment]
+1 for this feature, which is sorely lacking.

Thanks for this awesome application.

**Comment by lephuongbg** [comment]
@anusree-bruno My 5 cents on the least breaking way to implement this, due to how bruno currently work:

+ In the New Request dialog, only show "Type" selector. Request Name will simply be titled "Untitled Request". File name can have a unique random string in it to avoid collision.
+ The newly created .bru file through this new dialog will have an attribute `is_scratch = true`, and this flag will be cleared when the request is first saved.
+ When the request tab or bruno is closed, if `is_scratch = true`, then the .bru file is also removed.

**Comment by anusree-bruno** [comment]
Hi all! 
We’re actively working on this and wanted to share the flow we’re planning. We’re mainly trying to solve two things:
Making it quicker and simpler to create a new request, and
an option to create requests that aren’t tied to any collection (great for quick testing or temporary stuff).

Here’s the flow:

**A simpler “New Request” flow**:
Hovering over the “+” will show quick options for different request types.
Choosing one instantly creates a new Untitled request (untitled.bru, auto-incremented).
The context menu will also include these same quick shortcuts.

**Requests without a collection (scratch mode)**:
You’ll be able to create a request outside any collection for quick try-outs.
It stays temporary until you decide to save it.
When you hit save, Bruno will prompt you to choose which collection you want to put it in.

Would love to hear what you think!

**Comment by SamRay1024** [comment]
Hi @anusree-bruno ! Sounds good !
Thanks for this update.

**Comment by tomdapchai** [comment]
Hi @anusree-bruno , is there an expected version when this feature is rolled out? Very exciting for this one :D

**Comment by anusree-bruno** [comment]
We’re doing this step by step. The new request flow will be available in v3 (December), and scratch requests will follow in a later release (most likely in January).

**Comment by tomdapchai** [comment]
Thank you for the update!

**Comment by anusree-bruno** [comment]
Hi all, in the latest release clicking the + icon will immediately create a new request (defaults to HTTP and automatically picks up the request type and URL from the preset). If you want to choose a specific request type, right-clicking the + icon will show the available request types to select from.

**Comment by leverglowh** [comment]
I can confirm that it works as intended!

A couple of considerations, not sure if they belong here or in a new issue:
1. would it be possible to add a `cmd + T` shortcut for new tab that creates a new draft request?
2. would it be possible to have the focus on the url input when a new draft request is added?
3. double click on the + icon results in "untitled1 already exists" error

**Comment by chirag-bruno** [comment]
Hi @leverglowh, thanks for reviewing the feature and sharing your feedback. We really appreciate it.

To address your points:

* We’re planning to add key bindings, but we’re still evaluating what makes the most sense. Since we’ll soon support draft requests both within a collection and outside collections, we’re considering something along the lines of `Meta + N` and `Meta + Shift + N`, but this isn’t finalized yet.
* Agreed. This is already on our radar and should be included in one of the upcoming releases.
* The issue occurs when the `+` button is clicked twice before the initial file creation completes. We’ll address this by preventing double click handling on that button to avoid duplicate triggers.

Thanks again for the thoughtful suggestions.

**Comment by anusree-bruno** [comment]
Hi everyone,

We’ve added an option to create a request without selecting a collection. You can now create a request in your workspace using the + icon, and then choose to create a new collection or save it to an existing one when saving.
https://github.com/user-attachments/assets/c1af6010-e857-4d73-929a-8a1c8cf1d06c

I am closing this issue as resolved





> I can confirm that it works as intended!
> 
> A couple of considerations, not sure if they belong here or in a new issue:
> 
> 1. would it be possible to add a `cmd + T` shortcut for new tab that creates a new draft request?
> 2. would it be possible to have the focus on the url input when a new draft request is added?
> 3. double click on the + icon results in "untitled1 already exists" error

Point 1 & 2 will be addressed in an upcoming release.
